### DISCUSSION TITLE
Thought experiment: could FLARE run Rogue / Moria / MAngband / DiabloRL ?

### DISCUSSION THREAD

**Discussion by Des-Nerger** [flareteam/flare-game#955] [Ideas]
@dorkster, what do you think; using externally run / developed tools for random dungeon generation and things like that, could it possible in principle to reproduce the joy of playing the games like Rogue (1980), Moria (1983), MAngband (1997; a real-time roguelike!) or DiabloRL (2005), trying to run / develop them using flare-engine?

Flare-game, as it is now, seriously lacks in replayability, when compared to Diablo I-II. The obvious thing to do to fix this is to try to bring the roguelike part of Diablo back to flare-game. Which maybe isn't a trivial thing to do at first... But running a classic Unix roguelike on top of flare-engine seems like a thing approachable enough for me to try, even right now...

*// Out of curiousity I tried to build the [ludum-dare-31](https://github.com/dorkster/ludum-dare-31) game of yours to see if it looked any similar to a true Unix roguelike, but there were errors building it on a recent Debian, so I never run / played it.*

**Comment by dorkster** [comment]
Flare is already pretty close to being able to produce a rogue-like experience. We have the ability to generate loot in procedural way, and we can scale everything based on player level. We're really only missing a way to do procedural map generation. I'd argue implementing such a thing would be less difficult than shoehorning an existing game into flare-engine.

As for my personal opinion, I feel like too many games are trying to be "forever" games that encourage their players to replay them infinitely. I've fallen into the grasps of many myself. I'm fine with making a game that has finite playability.

(My Ludum Dare game should be fixed now. The include method for SDL_mixer changed since the time I wrote it. As for the game, don't expect much. It was made in 48-hours, though I did go back and make some improvements after the competition. If you're still curious, I also did a small [write-up](https://dorkster.github.io/nomans_dungeon.html) about it.)

**Reply by Blueberryy** [reply]
Hell yeah, open-source action RPG vampire survivor game would be amazing if that's possible!

**Reply by shinobody** [reply]
I think procedural mapgen would be great if only because it's a common feature in a lot of ARPGs. So if Flare wants to be the ARPG open source engine, I think it'd make sense to be at least able to do what Diablo did years ago.
